src/encoded/audit/file.py

```diff
@@ -513,6 +513,53 @@ def audit_file_matching_md5sum(value, system):
     yield AuditFailure('matching md5 sums', detail, level='WARNING')
 
 
+def audit_file_index_of(value, system):
+    '''
+    Files with output_type: index reads may specify that they are index_of
+    either 1 SE or 2 PE files. The SE/PE files must be from the same
+    experiment as index, other combinations are flagged as ERROR.
+    https://encodedcc.atlassian.net/browse/ENCD-5252
+    '''
+    if value['output_type'] != 'index reads':
+        return
+    if 'index_of' in value:
+        index_exp = value['dataset']['@id']
+        fastq_ids = set()
+        exp_ids = set()
+        indexed_files = 0
+        error_files = 0
+        run_type = set()
+        for each in value['index_of']:
+            fastq_exp = each['dataset']
+            indexed_files += 1
+            run_type.add(each['run_type'])
+            if index_exp != fastq_exp:
+                fastq_ids.add(each['@id'])
+                exp_ids.add(each['dataset'])
+                error_files += 1
+        if error_files >= 1:
+            fastq_ids_links = [audit_link(path_to_text(m), m) for m in fastq_ids]
+            exp_ids_links = [audit_link(path_to_text(n), n) for n in exp_ids]
+            detail = ('Index file {} is from experiment {}, but is used as an index for file(s) {} from experiment(s) {}'.format(
+                                audit_link(path_to_text(value['@id']), value['@id']),
+                                audit_link(path_to_text(value['dataset']['@id']), value['dataset']['@id']),
+                                ', '.join(fastq_ids_links),
+                                ', '.join(exp_ids_links)
+                                )
+                    )
+            yield AuditFailure('incorrect index file', detail, level='ERROR')
```

Comment:
for all three audit failures you created here you gave different audit categories (incorrect index file, mismatched index file, fastq missing index) while it seems more clear - it also adds 3 entries to already very long list of error audits facet - I think you should make one category ("inconsistent index reads file") or something like that and then the details could be different under it depending on the specific case, but all of them will fall under the same Audit category in the facet.

Reply:
All are now under 'inconsistent index file'